Module 1 - Day 3: Setup and mobile first

lmunozm1702 opened the pull request:
The pull request includes:

Repository setup.
Linters configuration.
index.html with navbar and main content for mobile layout.
style.css with design attributes and flex behavior.

lmunozm1702 committed:
Setup repository

Linters configuration, create:
.hintrc
.stylelintrc.json
.github/workflows/linters.yml

Create README.md

lmunozm1702 committed:
Create mobile main page

Create:
 index.html -> includes toolbar section
 /assets/css/styles.css

Add my_logo.png into assets/images folder

lmunozm1702 committed:
Add main content

Add main content into index.html
Add styles and flex behaivor into styles.css

Create .gitignore file

lmunozm1702 committed:
Update styles.css (STYLEINT issues)

Corrections for STYLEINT linter issues:

assets/css/style.css
   5:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
   7:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  18:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  39:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  49:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  49:16  ✖  Unexpected missing generic font family     font-family-no-missing-generic-family-keyword
  56:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  60:1   ✖  Expected empty line before rule            rule-empty-line-before
  65:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  65:16  ✖  Unexpected missing generic font family     font-family-no-missing-generic-family-keyword
  72:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  82:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
  82:16  ✖  Unexpected missing generic font family     font-family-no-missing-generic-family-keyword
  90:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
 101:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before
 104:3   ✖  Unexpected empty line before declaration   declaration-empty-line-before

Error: Process completed with exit code 2.

GiuseppeTG commented on index.html:
```diff
@@ -0,0 +1,43 @@
+<body>
+    <div class="container">
+        <header class="toolbar-dark-compact">
+            <div class="logo-placeholder"><img src="assets/images/my_logo.png" alt="Portal logo"></div>
```
- [x] Kindly add an `a` tag to your logo. This is for three reasons: convention, semantics, and user experience(UX). [Check this link](https://ux.stackexchange.com/questions/81727/why-is-it-standard-for-a-website-logo-to-navigate-to-the-home-page) for more info. 👍

lmunozm1702 replied:
It's done

GiuseppeTG commented on index.html:
```diff
@@ -0,0 +1,43 @@
+    <div class="container">
+        <header class="toolbar-dark-compact">
+            <div class="logo-placeholder"><img src="assets/images/my_logo.png" alt="Portal logo"></div>
+            <div class="normal-button"><i class="fa-solid fa-bars"></i></div>
```
- [x] Please wrap your hamburger menu with a `nav` tag. The `nav` tag is for semantics reasons and the right position for convention and user experience for mobile devices. 🤓

lmunozm1702 replied:
It's done.

GiuseppeTG commented on index.html:
```diff
@@ -0,0 +1,43 @@
+            <div class="normal-button"><i class="fa-solid fa-bars"></i></div>
+        </header>
+
+        <section class="main-content">
```
- [x] This section should contain the same background image as in the template. If an image is not part of the content (aesthetic reasons) use CSS background-image. [Check this link](https://stackoverflow.com/questions/492809/when-to-use-img-vs-css-background-image) for more info.

lmunozm1702 replied:
It's done!

GiuseppeTG commented on index.html:
```diff
@@ -0,0 +1,43 @@
+        </header>
+
+        <section class="main-content">
+            <h1>I’m Mariane Glad to see you!</h1>
```
[ OPTIONAL] You can put your own name here. This goes also for the logo 😸

lmunozm1702 replied:
It's done

GiuseppeTG commented on index.html:
```diff
@@ -0,0 +1,43 @@
+
+        <section class="main-content">
+            <h1>I’m Mariane Glad to see you!</h1>
+            <h5>I’m a software developer! I can help you build a product , feature or website Look through some of my
```
- [x] This paragraph should be a `p` tag. This way will be semantically correct and the screen readers and other technologies will thank you. 👍

lmunozm1702 replied:
It's done.

GiuseppeTG commented on index.html:
```diff
@@ -0,0 +1,43 @@
+                contact me.</h5>
+            <p class="social-media-title">LET'S CONNECT</p>
+
+            <ul class="social-media-icons">
+                <li><i class="fa-brands fa-twitter"></i></li>
+                <li><i class="fa-brands fa-linkedin-in"></i></li>
+                <li><i class="fa-brands fa-github"></i></li>
+            </ul>
```
- [x] You are missing some of the social media icons. Please add them and make sure they have an `a` tag.  Apart from being semantically correct, the `a` tag will let you connect to other sections of the website, using href. 🤓

lmunozm1702 replied:
Hi Guiseppe, please help with this... I don't know the social with the big "M" and the "hand peace" icons. Can you tell me the names of them? please?

lmunozm1702 replied:
It's done.

GiuseppeTG reviewed (changes requested):
Hi @lmunozm1702  👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![Alt Text](https://media0.giphy.com/media/wA6aYAO3jH6RmUIBYs/giphy.gif)

## Highlights
✅ Good commit history and documentation
✅ Nice implementation of flex
✅ Professional PR 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

GiuseppeTG commented on assets/css/style.css:
```diff
@@ -0,0 +1,101 @@
+.container {
+  max-width: 375px;
+  height: 652px;
+  border: 1px grey solid;
```
- [x] Don't forget to take out this border as it is not in the template. I think it was used as a guide, which is ok 😺

lmunozm1702 replied:
It's done, the border was used just to reference the size of the mobile.

lmunozm1702 committed:
Code Review: Add <a> tag to logo

Update index.html to add an <a> tag to the logo.

lmunozm1702 committed:
Code Review: add <nav> tag

Wrap the "hamburguer menu" with a <nav> tag

lmunozm1702 committed:
Code Review: set background image

Add /assets/images/header-shapes-mobile.png into project folder.

Update index.html and /assets/css/style.css to set the main section background image.

lmunozm1702 committed:
Code Review: Change name to real.

Update index.html to set my real name into the text.

lmunozm1702 committed:
Code Review: change <H5> by a <P> tag

Update index.html and style.css to change the <h5> tag into the main content by a <p> tag.

lmunozm1702 committed:
Code Review: Missed social icons

Update index.html and styles.css to add 2 missed icons and set an <a> tag to link the social media sites.

lmunozm1702 committed:
Code Review: container border remove

Updated styles.css to remove (comment), the border of container, it was used how size reference.

lmunozm1702 committed:
Code Review: STYLELINT issues

Updates styles.css to resolve:

assets/css/style.css
 8:3   ✖  Expected empty line before comment   comment-empty-line-before
 8:5   ✖  Expected whitespace after "/*"       comment-whitespace-inside
 8:27  ✖  Expected whitespace before "*/"      comment-whitespace-inside

lmunozm1702 committed:
Code Review: STYLEINT issue

Update style.css to remove border in container class.

assets/css/style.css
 8:3   ✖  Expected empty line before comment   comment-empty-line-before
 8:5   ✖  Expected whitespace after "/*"       comment-whitespace-inside
 8:27  ✖  Expected whitespace before "*/"      comment-whitespace-inside

victorokoroji reviewed (approved):
Hi @lmunozm1702, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 
<img src="https://media3.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif"/>

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

lmunozm1702 commented:
Great news for me!!!! thanks a lot!!!
